spyder/app/mainwindow.py

```diff
@@ -119,12 +119,20 @@
 #==============================================================================
 from spyder.utils.qthelpers import qapplication, MENU_SEPARATOR
 from spyder.config.base import get_image_path
+from spyder.config.gui import is_dark_interface
 MAIN_APP = qapplication()
 
 if PYQT5:
-    APP_ICON = QIcon(get_image_path("spyder.svg"))
+    if is_dark_interface():
+        icon_filename = "spyder.svg"
+    else:
+        icon_filename = "spyder_dark.svg"
 else:
-    APP_ICON = QIcon(get_image_path("spyder.png"))
+    if is_dark_interface():
+        icon_filename = "spyder.png"
+    else:
+        icon_filename = "spyder_dark.png"
+APP_ICON = QIcon(get_image_path(icon_filename))

```

Comment:
@dalthviz, I think this is not the right place to do this because this is the icon used in the dock bar on macOS and Linux. Please see PR #4443 for the work that implemented this.

So please move the selection of a dark or light icon to the creation of the *About Spyder* dailog itself.